### ISSUE TITLE
minishift-release job fails to add release notes for it is not able to get the milestone Id

### ISSUE THREAD

**Issue by anjannath** [minishift/minishift#2186]
https://github.com/minishift/minishift/blob/e07e789e75854fc4890ceca96c7cb683b516c0d6/centos_ci.sh#L229-L249

**Comment by anjannath** [comment]
Github api is acting weired, it is not returning the full list of milestones:

```
[redhat@localhost minishift]$ curl -s https://api.github.com/repos/minishift/minishift/milestones?state=all  | jq -r '.[] | .title'
v0.1.1
v0.3.0
v0.3.4
v2.0.0
v1.4.1
v1.17.0
v1.18.0
v1.19.0
v1.15.1
v1.0.0-beta.1
v1.0.0-beta.2
v1.0.0-beta.3
v1.0.0-beta.4
v1.0.0-beta.5
v1.0.0-rc.1
v1.0.0-rc.2
v1.0.1
v1.1.0
v1.2.0
v1.3.0
v1.3.1
v1.4.0
v1.5.0
v1.6.0
v1.7.0
v1.8.0
v1.9.0
v1.10.0
v1.11.0
v1.12.0
```

**Comment by anjannath** [comment]
This was seen during the release job of v1.15.0 but then during v1.15.1 release job.. the release notes got updated..  :-/

trying to fetch all the milestones from github again, we get the v1.15.1 milestone listed but not v.1.15.0

```
[redhat@dhcp35-10 minishift]$ curl -s https://api.github.com/repos/minishift/minishift/milestones?state=all  | jq -r '.[] | .title'
v0.1.1
v0.3.0
v0.3.4
v2.0.0
v1.4.1
v1.17.0
v1.18.0
v1.19.0
v1.15.1 <---
v1.0.0-beta.1
v1.0.0-beta.2
v1.0.0-beta.3
v1.0.0-beta.4
v1.0.0-beta.5
v1.0.0-rc.1
v1.0.0-rc.2
v1.0.1
v1.1.0
v1.2.0
v1.3.0
v1.3.1
v1.4.0
v1.5.0
v1.6.0
v1.7.0
v1.8.0
v1.9.0
v1.10.0
v1.11.0
v1.12.0
```

**Comment by budhrg** [comment]
> This was seen during the release job of v1.15.0 but then during v1.15.1 release job.. the release notes got added successfully.. :-/

This is little weird.

**Comment by praveenkumar** [comment]
Call which is failing to get milestone was below looks like we need to add direction to get it properly.

```
$ MILESTONE_ID=`curl -s https://api.github.com/repos/minishift/minishift/milestones?state=all  | jq --arg version "v1.18.0" -r ' .[] | if .title == $version then .number else empty end'`

$ echo $MILESTONE_ID
<Null>

$ MILESTONE_ID=`curl -s 'https://api.github.com/repos/minishift/minishift/milestones?state=all&direction=desc'  | jq --arg version "v1.18.0" -r ' .[] | if .title == $version then .number else empty end'`

11:19 $ echo $MILESTONE_ID
35
```

**Comment by praveenkumar** [comment]
We need pagination for this.

```
$ curl -s 'https://api.github.com/repos/minishift/minishift/milestones?state=all&page=1&per_page=150'  | jq -r '.[] | .title'
v0.1.1
v0.3.0
v0.3.4
v2.0.0
v1.4.1
v1.15.1
v1.20.0
v1.21.0
v1.0.0-beta.1
v1.0.0-beta.2
v1.0.0-beta.3
v1.0.0-beta.4
v1.0.0-beta.5
v1.0.0-rc.1
v1.0.0-rc.2
v1.0.1
v1.1.0
v1.2.0
v1.3.0
v1.3.1
v1.4.0
v1.5.0
v1.6.0
v1.7.0
v1.8.0
v1.9.0
v1.10.0
v1.11.0
v1.12.0
v1.13.0
1.13.1
v1.14.0
v1.15.0
v1.0.0
v1.16.0
v1.16.1
v1.17.0
v1.18.0
v1.19.0
```

**Comment by LalatenduMohanty** [comment]
Resolved via #2416
